ui/jobs/buff_tracker.ts

```diff
@@ -25,12 +69,12 @@ export class Buff {
     this.cooldownSortKeyBase = 2000;
   }
 
-  addCooldown(source, effectSeconds) {
+  addCooldown(source: string, effectSeconds: number): void {
     if (!this.info.cooldown)
       return;
     if (this.cooldown[source]) {
       // Unexpected use of the same cooldown by the same name.
-      this.cooldown[source].removeCallback();
+      this.cooldown[source]?.removeCallback();
```

Comment:
I think you could remove `if (this.cooldown[source]) {` check before this if you're using optional chaining.  Maybe the comment could be "// Remove any preexisting cooldowns with the same name in case they unexpectedly exist" instead.

Same thing for the ready check below.

Reply:
You're right, thanks for catching this.